source/geometries/Next100Vessel.cc

```diff
@@ -120,112 +124,166 @@ namespace nexus {
-    G4ThreeVector tracking_flange_pos(0, 0, flange_z_pos_);
-    G4ThreeVector energy_flange_pos(0, 0, -1. * flange_z_pos_);
+    G4double endcap_z_pos = (body_length_ / 2.) + endcap_in_z_width_ - endcap_in_rad_;
+    G4ThreeVector energy_endcap_pos  (0, 0,  endcap_z_pos);
+    G4ThreeVector tracking_endcap_pos(0, 0, -endcap_z_pos);
+    G4ThreeVector energy_flange_pos  (0, 0, flange_ep_z_pos);
+    G4ThreeVector tracking_flange_pos(0, 0, flange_tp_z_pos);
 
-    // Body + Tracking endcap
-    G4UnionSolid* vessel_solid = new G4UnionSolid("VESSEL", vessel_body_solid, vessel_tracking_endcap_solid,
-						  0, tracking_endcap_pos);
+    // tracking endcap z --> -z (we can rotate either in X or Y)
+    G4RotationMatrix* xRot = new G4RotationMatrix;
+    xRot->rotateX(180. * deg);
 
-    // Body + Tracking endcap + Energy endcap
-    vessel_solid = new G4UnionSolid("VESSEL", vessel_solid, vessel_energy_endcap_solid,
-				    0, energy_endcap_pos);
+    // Body + Energy endcap
+    G4UnionSolid* vessel_solid = new G4UnionSolid("VESSEL", vessel_body_solid, vessel_endcap_solid,
+						  0, energy_endcap_pos);
 
-    // Body + Tracking endcap + Energy endcap + Tracking flange
-    vessel_solid = new G4UnionSolid("VESSEL", vessel_solid,
-				    vessel_flange_solid, 0, tracking_flange_pos);
+    // Body + Energy endcap + Tracking endcap
+    vessel_solid = new G4UnionSolid("VESSEL", vessel_solid, vessel_endcap_solid,
+				    xRot, tracking_endcap_pos);
 
-    // Body + Tracking endcap + Energy endcap + Tracking flange + Energy flange
+    // Body + Energy endcap + Tracking endcap + Energy flange
     vessel_solid = new G4UnionSolid("VESSEL", vessel_solid,
-				    vessel_flange_solid, 0, energy_flange_pos);
+				    vessel_ep_flange_solid, 0, energy_flange_pos);
 
-    // Adding nozzles
-    vessel_solid = new G4UnionSolid("VESSEL", vessel_solid, small_nozzle_solid,
-				    0, G4ThreeVector(0., up_nozzle_ypos_, 0.) );
-    vessel_solid = new G4UnionSolid("VESSEL", vessel_solid, large_nozzle_solid,
-				    0, G4ThreeVector(0., central_nozzle_ypos_, 0.) );
-    vessel_solid = new G4UnionSolid("VESSEL", vessel_solid, large_nozzle_solid,
-				    0, G4ThreeVector(0., down_nozzle_ypos_, 0.) );
-    vessel_solid = new G4UnionSolid("VESSEL", vessel_solid, small_nozzle_solid,
-				    0, G4ThreeVector(0., bottom_nozzle_ypos_, 0.) );
-
-
-
-    // Body gas + Tracking endcap gas
+    // Body + Energy endcap + Tracking endcap + Energy flange + Tracking flange
+    vessel_solid = new G4UnionSolid("VESSEL", vessel_solid,
+				    vessel_tp_flange_solid, 0, tracking_flange_pos);
+
+    // Add ports (x,y at 45*deg)
+    G4double port_x = (vessel_out_rad+port_base_height/2.)/sqrt(2.);
+    G4double port_y = port_x;
+    port_gas_x_ = (vessel_in_rad_+port_base_height/2.+vessel_thickness_/2.)/sqrt(2.);
+    port_gas_y_ = port_gas_x_;
+
+    G4double port_reference_z = -body_length_/2. + endcap_in_body_;
+    port_z_1a_ = port_reference_z +   702. * mm;
```

Comment:
I would add at some point in the 'Next100Vessel.cc' file which port is each one (or at least which side is a or b) with comments, not only in the PR document. It's not intuitive the way they are defined right now. Also, I wonder if we should numerate them with respect to their distant to the TP.